packages/client-python/src/rocketride/core/runtime/downloader.py

```
@@ -0,0 +1,82 @@
+                    async for chunk in resp.content.iter_chunked(8192):
+                        f.write(chunk)
+                # fd is now closed, don't close again in finally
+                tmp_fd = -1
+
+        # Extract
+        dest = engines_dir(version)
+        dest.mkdir(parents=True, exist_ok=True)
+
+        if asset.endswith('.tar.gz'):
+            with tarfile.open(tmp_path, 'r:gz') as tar:
+                tar.extractall(path=str(dest))
```

Ruff (0.15.7)

[error] Uses of `tarfile.extractall()`

(S202)